lib/HeroBonus.h

```diff
@@ -331,6 +330,21 @@ class DLL_LINKAGE CAddInfo : public std::vector<si32>
 	BONUS_NAME(SPECIAL_FIXED_VALUE_ENCHANT) /*specialty spell like Melody has, constant spell effect (i.e. 3 luck), additionalInfo = value to fix.*/\
 	BONUS_NAME(TOWN_MAGIC_WELL) /*one-time pseudo-bonus to implement Magic Well in the town*/\
 	BONUS_NAME(LIMITED_SHOOTING_RANGE) /*limits range of shooting creatures, doesn't adjust any other mechanics (half vs full damage etc). val - range in hexes, additional info - optional new range for broken arrow mechanic */\
+	BONUS_NAME(LEARN_BATTLE_SPELL_CHANCE) /*skill-agnostic eagle eye chance. subtype = 0 - from enemy, 1 - TODO: from entire battlefield*/\
+	BONUS_NAME(LEARN_BATTLE_SPELL_LIMIT) /*skill-agnostic eagle eye limit, subtype - school (-1 for all), others TODO*/\

```

Comment:
Maybe "LEARN_BATTLE_SPELL_LEVEL_LIMIT"?

Reply:
@IvanSavenko maybe consolidate all level limits to one bonus with 3 subtypes (Wisdom, Scholar and Eagle Eye)?

Reply:
I'd rather not. Too different meaning. Subtypes should act as filter/more specific behavior. Completely different effects should have separate bonuses.

If I were to change these bonuses I would rather give all of them magic school subtype

Reply:
@IvanSavenko It is planned, but not in this PR, this is a simple conversion for now. Extending spell effects to generic framework will be done later.

Found 2 weird issues with global effects:
1. Heroes in tavern shows with 0/0 mana (recruited normal)
2. Starting heroes have 1 mana (probably because they have 0 mana on map, and we do not init 0 mana).

Reply:
I guess heroes from taverns are not attached to bonus system.
Once you recruit them, they get attached to player node. But not while in tavern...

One option would be to attach them to global node directly (and remove on recruit).
Another would be to implement base mana per knowledge some other way.

Reply:
@IvanSavenko will try. 

About globals - I think it is a preparation to per-map setting of globals (and allows eg. arena maps with globals changed).

Reply:
@IvanSavenko bonus renamed. Will look for mana issue fix.

Reply:
mana issues fixed too.